File: 026.RemoveDuplicatesFromSortedArray/clywin123/removeDuplicates.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeDuplicates(self, nums):
        if(not nums):
            return 0
        pointer = 1
        previous = nums[0]
        for i in range(1,len(nums)):
            if(nums[i]!=previous and nums[i]>previous):
                previous = nums[i]
                nums[i],nums[pointer] = nums[pointer],nums[i]
                pointer +=1
        #print nums[:pointer]
        return pointer

File: 080.RemoveDuplicatesfromSortedArray-II/wyrj/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let newLen = 2;
    if (nums.length < newLen) return nums.length;
    let skip = nums[0] === nums[1];
    for (let i = 2, len = nums.length; i < len; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[newLen] = nums[i];
            newLen += 1;
            skip = false;
        } else if (!skip) {
            nums[newLen] = nums[i];
            newLen += 1;
            skip = true;
        }
    }
    return newLen;
};
